usefulUtilities/voteApp/appResources/voteApp.js

```diff
@@ -0,0 +1,1022 @@
+        GOTO_DOMAIN = CONFIG.GOTO_DOMAIN,
+        VOTE_AVATAR = CONFIG.VOTE_AVATAR,
+        VOTE_DOMAIN = CONFIG.VOTE_DOMAIN;
+
+    // Google scripts type events
+    var VOTE_GOOGLE = "vote",
+        GET_INFO_GOOGLE = "getInfo";
+
+    // Vue UI update event
+    var UPDATE_UI = EVENT_NAME + "_update_ui"; // !important must match Tablet.js
+    
+    // Static strings
+    var DOMAIN = "domain";
+    var AVATAR = "avatar";
+    
+    var DEBUG = true;
+
+    // Local Settings.setValue/getValue key
+    var VOTE_APP_SETTINGS_NAME = EVENT_NAME + "_voteApp"; // "Futvrelands_11_17_2018_voteApp"
+    
+    var DEFAULT_VOTE_APP_SETTINGS = {
+        visited: false,
+        domains: [],
+        voted: {
+            avatar: "", domain: ""
+        }
+    };
+
+    // UI variables
+    var ui;
+    var dataStore = {
+        unload: false,
+        loading: true,
+        loggedin: true,
+
+        visited: false, // set by zone HOLIDAY APP
+
+        voted: {
+            domain: false,
+            avatar: false
+        },
+        openPolls: {
+            avatar: false,
+            domain: false
+        },
+        visitedAllDomains: false, // for domains
+        domains: [],
+        avatars: []
+    };
+    
+
+    var domains = {
+        DOMAIN: "domain",
+        domainsInfo: {},
+        sendVote: function (name) {
+
+            var _this = this;
+
+            if (DEBUG) {
+                print("Domains sendVote");
+            }
+
+            _this.domainsInfo[name.toLowerCase()].voted = true;
+            dataStore.voted.domain = true;
+
+            if (DEBUG) {
+                print("Domains sendVote voted name", JSON.stringify(dataStore.voted.domain));
+            }
+
+            updateUI();
+
+            var params = utils.encodeURLParams({
+                type: VOTE_GOOGLE,
+                time: Date.now(),
+                uuid: MyAvatar.sessionUUID,
+                username: AccountServices.username,
+                name: name,
+                eventName: EVENT_NAME,
+                eventDate: EVENT_DATE,
+                contestName: DOMAIN
+            });
+            var onComplete = function (request) {
+
+                var voteAppSettings = utils.getVoteAppSettings();
+                voteAppSettings.voted[DOMAIN] = name.toLowerCase();
+                Settings.setValue(VOTE_APP_SETTINGS_NAME, voteAppSettings);
+
+                if (DEBUG) {
+                    print("Domain sendVote onComplete", JSON.stringify(voteAppSettings));
+                }
+
+            };
+            var onError = function () {
+                if (DEBUG) {
+                    print("Error in VoteApp.js: Issue in sendDomainVote()");
+                }
+
+                _this.domainsInfo[name.toLowerCase()].voted = false;
+                dataStore.voted.domain = false;
+
+                updateUI();
+            };
+
+            utils.sendRequest(GOOGLE_SCRIPTS_URL, params, onComplete, onError);
```

Comment:
I'd like to see this using the `request` module rather than `utils.sendRequest`. it would greatly simplify the above code